src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/prevalue/blocklist.blockconfiguration.controller.js

```diff
@@ -188,28 +165,51 @@
             $scope.model.value.push(blockType);
 
             vm.openBlockOverlay(blockType);
-
         };
 
-        vm.openBlockOverlay = function (block) {
+        vm.openBlockOverlay = ($event, block) => {
+            
+            // bit hacky, but this catches bubbled events triggered
+            // by a keypresson the delete button
+            if ($event.originalEvent.target.classList.contains('__action'))
+                return false;
 
             var elementType = vm.getElementTypeByKey(block.contentElementTypeKey);
 
             if(elementType) {
-                localizationService.localize("blockEditor_blockConfigurationOverlayTitle", [elementType.name]).then(function (data) {
+                localizationService.localize("blockEditor_blockConfigurationOverlayTitle", [elementType.name]).then(data => {
 
                     var clonedBlockData = Utilities.copy(block);
                     vm.openBlock = block;
 
-                // open property settings editor
-                editorService.open(overlayModel);
-            });
+                    var overlayModel = {
+                        block: clonedBlockData,
+                        title: data,
+                        view: "views/propertyeditors/blocklist/prevalue/blocklist.blockconfiguration.overlay.html",
+                        size: "small",
+                        submit: overlayModel => {
+                            loadElementTypes()// lets load elementType again, to ensure we are up to date.
+                            TransferProperties(overlayModel.block, block);// transfer properties back to block object. (Doing this cause we dont know if block object is added to model jet, therefor we cant use index or replace the object.)
```

Comment:
@nathanwoulfe I know this isn't specific related to the recent changes, but I think this function `TransferProperties()` should use camel case like `transferProperties()`. I also wonder if this couldn't be placed in a helper/service.
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/prevalue/blocklist.blockconfiguration.controller.js#L13-L17

Or maybe we can just use `Utilities.copy(src, dst)` function?
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/utilities.js#L21

Reply:
Hi Bjarne

I think that Utilities.copy is doing a deep clone. Its important that we don't do a deep copy here, as we want the references to float over.

The reason for it to be with a capitalized first letter is that it's a Static method. As we do with Classes etc.

I think you should hold back a bit with all these micro-optimizations, as it becomes hard to track the code backwards when the code has been changed just for styling purposes. I'm mostly concerned about the changing of the methods to arrow functions without any other change or reason to do so.

Reply:
@nielslyngsoe okay, makes sense. I haven't looked further at the differences on the "block" object and the "blockConfiguration" object, but I guess this object is simpler.

Anyway I was wondering if there isn't anywhere else in code, where we also do a "flat" copy as well and therefore this might be something to re-use.

Reply:
@bjarnef would be nice to investigate. Could for sure be useful in other places so refactoring it into a utility method would be nice.

Reply:
@nielslyngsoe sure, but for now I think @nathanwoulfe added a workaround in this PR, to fix the annoying issue, where keypress on block delete button both trigger delete overlay and edit overlay 😅🙈

Reply:
@bjarnef Oh, yes. We should still have that. :-) But we should in general try to keep one PR for each feature/bug. So we should extract that anyway. That would make @nul800sebastiaan happy. Because we use PRs/Issues to get an overview of the changes and fixes made in a release.
In this way, it's also easier to later return to a feature/bug-fix, evolve it, pull it for a hotfix in an earlier version for a patch releases. Or eventually to revert it if it causes trouble. :-) I hope that makes sense :-)